manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsService.java

```diff
@@ -9,11 +10,9 @@
 
 public interface ConnectorsService {
 
-    Optional<ConnectorEntity> createConnectorIfNeeded(BaseAction resolvedAction,
+    Optional<ConnectorEntity> persistConnectorIfNeeded(BaseAction resolvedAction,
             Processor processor,
             ActionProvider actionProvider);
 
-    void deleteConnectorIfNeeded(BaseAction resolvedAction,
-            Processor processor,
-            ActionProvider actionProvider);
+    List<ConnectorEntity> deleteConnectorIfNeeded(Processor processor);
```

Comment:
@lucamolteni How comes a create returns a single `ConnectorEntity`, but a delete returns a list of `ConnectorEntity` ? Is there a scenario where the there is a one-to-many relationship between `Processor` and `Connector`?

Reply:
It's a `OneToOne` mapped as a `OneToMany/ManyToOne`, we cannot exclude that for a bug multiple connectors where created, so it's safer to delete them all anyway